lib/index.js

```diff
@@ -46,187 +43,103 @@ Ramen.prototype.initialize = function() {
  */
 
 Ramen.prototype.loaded = function() {
-  return is.object(window.Ramen);
+  return !!(window._ramen && window._ramen.push !== Array.prototype.push);
 };
 
-/**
- * Identify.
- *
- * @api public
- * @param {Identify} identify
- */
+Ramen.prototype.group = function(group) {
+  if (!this.identified) {
+    this.identify({ anonymous: true });
+  }
 
-Ramen.prototype.identify = function(identify) {
-  privateGo({ integration: this, identify: identify });
-};
+  var props = group.properties();
+  props = alias(props, { createdAt: 'created' });
+  props = alias(props, { created: 'created_at' });
+  var id = group.groupId();
+  if (id) props.id = id;
 
-Ramen.prototype.group = function(group) {
-  privateGo({ integration: this, group: group });
+  window._ramen.group(props);
 };
 
-Ramen.prototype.page = function(page) {
-  privateGo({ integration: this, page: page });
+Ramen.prototype.page = function() {
+  if (!this.identified) {
+    this.identify({ anonymous: true });
+  }
+
+  window._ramen.page();
```

Comment:
question -- sometimes our users like to manually override the properties such as `url`, `title`, `path`, etc by pushing them inside the `.page()` call like

```
analytics.page({
   title: 'some custom title',
   path: 'some custom path'
});
```

Does Ramen allow you to override its own `.page()` call's auto parsing of the document by passing in arguments into the `window._ramen.page();`? If so we might want to allow support for that. But happy to make that a separate PR since this one already is a big overhaul :)